driver/labjack/reader_source.cpp

```diff
@@ -0,0 +1,480 @@
+    );
+    if(err != freighter::NIL){
+        LOG(ERROR) << "[labjack.reader] Error retrieving device: " << err.message();
+        return;
+    }
+
+    if(dev.model == "LJM_dtT4"){
+        this->reader_config.device_type = "T4";
+    } else if(dev.model == "LJM_dtT7"){
+        this->reader_config.device_type = "T7";
+    } else if(dev.model == "LJM_dtT8"){
+        this->reader_config.device_type = "T8";
+    } else {
+        LOG(ERROR) << "[labjack.reader] Unsupported device type: " << dev.model;
+        return;
+    }
+    this->init_stream();
+    this->init_tcs();
+}
+
+void labjack::ReaderSource::init_tcs(){
+    if(this->reader_config.tc_channels.empty()){
+        return;
+    }
+    {
+        std::lock_guard<std::mutex> lock(labjack::device_mutex);
+        if(check_err(LJM_Open(LJM_dtANY, LJM_ctANY, this->reader_config.serial_number.c_str(), &this->handle), "init_tcs.LJM_OPEN")){
+            LOG(ERROR) << "[labjack.reader] LJM_Open error";
+            return;
+        }
+    }
+
+    for(auto &channel : this->reader_config.channels){
+        if(channel.channel_type ==  "AI"){
+            this->check_err(
+                    LJM_eWriteName(
+                            this->handle,
+                            (channel.location + "_RESOLUTION_INDEX").c_str(),
+                            0
+                    ), "init_tcs.LJM_eWriteName.RESOLUTION_INDEX"
+            );
+            if(this->reader_config.device_type == "T7" || this->reader_config.device_type == "T8") {
+                this->check_err(
+                        LJM_eWriteName(
+                                this->handle,
+                                (channel.location + "_RANGE").c_str(),
+                                0
+                        ), "init_tcs.LJM_eWriteName.RANGE"
+                );
+            }
+            if(this->reader_config.device_type == "T7") {
+                this->check_err(
+                        LJM_eWriteName(
+                                this->handle,
+                                (channel.location + "_NEGATIVE_CH").c_str(),
+                                channel.neg_chan
+                        ), "init_tcs.LJM_eWriteName.NEGATIVE_CH"
+                );
+            }
+        }
+    }
+
+    int msDelay = 1000;
+    auto err = LJM_StartInterval(
+            this->handle,
+            msDelay * 1000
+    );
+
+    for(auto &channel : this->reader_config.tc_channels){
+        if(this->reader_config.device_type == "T4"){
+            LOG(ERROR) << "[labjack.driver] thermocouple channels not currently supported for T4 devices";
+            continue;
+        }
+        this->configure_tc_ain_ef(channel.tc_config);
+    }
+}
+
+void labjack::ReaderSource::init_stream(){
+    if(!this->reader_config.tc_channels.empty()){
+        return;
+    }
+    double INIT_SCAN_RATE = this->reader_config.sample_rate.value;
+    int SCANS_PER_READ = (int)INIT_SCAN_RATE / this->reader_config.stream_rate.value;
```

Comment:
Do you have an opinion on whether we should only use C++ style casts or are C style casts ok?

Reply:
we should use C++ style casts because the compiler will flag sus casts for C++ style casts that it wont for c style. ill replace with that